src/api/source/geojson.js

```diff
@@ -21,43 +46,59 @@ export default class GeoJSON extends Base {
     }
 
     requestData() {
-        // TODO: split in two functions: (metadata) / (data)
+        // TODO: split it in two functions: (metadata) / (data)
         //
         if (this._status === 'init') {
             this._status = 'metadata';
-            return new Promise(resolve => {
-                let metadata = {};
-                resolve(metadata);
-            });
+            return this._requestMetadata();
         } else if (this._status === 'metadata') {
             this._status = 'data';
-            const geometry = this._decodeGeometry();
-            const properties = {};
-            const dataframe = new Dataframe(
-                { x: 0, y: 0 },
-                1,
-                geometry,
-                properties,
-            );
-            dataframe.type = this._getDataframeType(this._type);
-            dataframe.active = true;
-            dataframe.size = this._features.length;
-            this._addDataframe(dataframe);
+            this._requestData();
+        }
+    }
+
+    _checkData(data) {
+        if (util.isUndefined(data)) {
+            throw new CartoValidationError('source', 'dataRequired');
+        }
+        if (!util.isObject(data)) {
+            throw new CartoValidationError('source', 'dataObjectRequired');
         }
     }
 
     _getFeatures(data) {
-        data = data || {};
         if (data.type === 'FeatureCollection') {
             return data.features || [];
         } else if (data.type === 'Feature') {
             return [data];
         }
         else {
-            throw Error('No valid GeoJSON data');
+            throw new CartoValidationError('source', 'nonValidGeoJSONData');
         }
     }
 
+    _requestMetadata() {
+        return new Promise(resolve => {
```

Comment:
```javascript 
return Promise.resolve({});
```

?

Reply:
This is a TODO. We will add metadata extraction in the future.